x-pack/plugins/threat_intelligence/public/common/navigation/security_solution_links.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TIPage, TIPageId } from './types';
+import { threatIntelligencePages } from './constants';
+
+/**
+ * Properties used in the Security Solution plugin to enable deep linking.
+ * The properties come from SecuritySolutionDeepLink (x-pack/plugins/security_solution/public/app/deep_links/index.ts).
+ *
+ * If we want to control more from within the Threat Intelligence plugin, we can keep growing this interface.
+ */
+interface TIDeepLink<TId extends string = TIPageId> {
```

Comment:
the next 3 interfaces are almost duplication of existing interfaces inside the Security Solution (SecuritySolutionDeepLink, LinkItem and NavTab).
In the future it could be worse extracting these into some reusable utils plugin/package, as the Cloud Security Posture plugin is doing something similar...